scipy/optimize/_minpack_py.py

```diff
@@ -659,7 +659,9 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         If the Jacobian matrix at the solution doesn't have a full rank, then
         'lm' method returns a matrix filled with ``np.inf``, on the other hand
         'trf'  and 'dogbox' methods use Moore-Penrose pseudoinverse to compute
-        the covariance matrix.
+        the covariance matrix. Covariance matrices with large condition numbers
+        (e.g. computed with`numpy.linalg.cond`) may indicate that results are
+        unreliable.
```

Comment:
I wonder if this would be more visible in the notes section, or if adding an example to the docstring where this happens may help indicate that this is expected if the matrix is ill-conditioned. Otherwise, looks good to me.

```suggestion
        the covariance matrix. Covariance matrices with large condition numbers
        (e.g. computed with `numpy.linalg.cond`) may indicate that results are
        unreliable.
```

Reply:
Thanks for finding that.
I thought about adding it in the notes, but I saw the note about infinite covariance, so I thought the two statements belonged together. I thought of the singular jacobian -> infinite covariance as the extreme case of this, in which case a nearly singular jacobian (I think) leads to a very high condition number.

I'll try to add an example though.

Reply:
@melissawm I added an example. How does that look?

Reply:
Looks great, thanks!